src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -102,8 +111,14 @@ def self.getModules(env)
       # NOTE: If in the future we want to collect more
       #       Module settings, this should be more programatic
       #       rather than getting these settings one by one
-      module_data[:name] = mod.forge_name
-      module_data[:version] = mod.version
+      if mod.forge_name
+        module_data[:name] = mod.forge_name
+        module_data[:version] = mod.version
+      else
+        # no metadata.json file, so all we have is `name`
+        module_data[:name] = mod.name
+        module_data[:version] = nil
```

Comment:
I was actually thinking of cleaning up the code here, so I'll go ahead and do that!